starter-bots/cplusplus/cppbot/cppbot.cpp

```diff
@@ -0,0 +1,106 @@
+#include <iostream>
+#include "./rapidjson/document.h"
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <functional>
+#include <cmath>
+#include <random>
+
+struct POINT
+{
+  int x;
+  int y;
+};
+
+enum class SurfaceObject : uint8_t {
+  EMPTY = 0,
+  MUD = 1,
+  OIL_SPILL = 2,
+  OIL_ITEM = 3,
+  FINISH_LINE = 4,
+  BOOST = 5
+};
+
+SurfaceObject GetObjectAt(rapidjson::Document& roundJSON, POINT point)
+{
+  for (rapidjson::Value::ConstValueIterator rowItr = roundJSON["worldMap"].Begin(); rowItr != roundJSON["worldMap"].End(); ++rowItr) {
+    for (rapidjson::Value::ConstValueIterator colItr = (*rowItr).Begin(); colItr != (*rowItr).End(); ++colItr) {
+
+        POINT thisPos{(*colItr)["position"]["x"].GetInt(), (*colItr)["position"]["y"].GetInt()};
```

Comment:
Declaring a `POINT` class here seems a bit unnecessary. If this is strictly a utility method, then I would suggest you inline the values in the if `if ((*colItr)["position"]["x"].GetInt() == point.x && (*colItr)["position"]["y"].GetInt() == point.y)`
I am just concerned because this is an extra stack allocation for this function and there is a limited stack size, so this might make some recursion based approaches bomb out more quickly that if the values were inlined